drivers/display/ssd1306.c

```diff
@@ -238,51 +236,36 @@ int ssd1306_write(const struct device *dev, const u16_t x, const u16_t y,
 		return -1;
 	}
 
-	if (x != 0U && y != 0U) {
+
+#if defined(CONFIG_SSD1306_DEFAULT)
+
+	if ((y & 0x7) != 0U) {
 		LOG_ERR("Unsupported origin");
 		return -1;
 	}
 
-#if defined(CONFIG_SSD1306_DEFAULT)
-	struct ssd1306_data *driver = dev->driver_data;
+	for (size_t pidx = y/8; pidx < (y+desc->height)/8; pidx++) {
+		if (ssd1306_write_page(dev, pidx,  x, buf,  desc->width)) {
```

Comment:
Will this implementation not result in much more data transfer than the original implementation?
I'm wondering if the commands could not be send once to setup the drawing region and then transfer all data.

Reply:
The original implementation can only transfer data from (0, 0)，Can't write the specified area within the framebuffer.
For ssd1306, If the width is not equal to 128 and height across several pages, it is not possible to continuously write image data of any size area to the framebuffer.  This is a hardware limitation of ssd1306, which causes it to be written multiple times by write page.
Of course, an area with a starting point of (0,0) and a width of 128 can also be transmitted at one time. However, this effect is not obvious for the improvement of transmission efficiency, but the code is complicated.